packages/compiler/test/shadow_css/keyframes_spec.ts

```diff
@@ -0,0 +1,498 @@
+        @keyframes steps {}
+        `;
+    const result = s(css, 'host-a');
+    expect(result).toContain('animation: inherit;');
+    expect(result).toContain('animation: unset;');
+    expect(result).toContain('animation: 3s ease reverse host-a_foo;');
+    expect(result).toContain('animation: 5s host-a_foo 1s backwards;');
+    expect(result).toContain('animation: none 1s host-a_foo;');
+    expect(result).toContain('animation: .5s host-a_foo paused;');
+    expect(result).toContain('animation: 1s running host-a_foo;');
+    expect(result).toContain('animation: 3s linear 1s infinite running host-a_foo;');
+    expect(result).toContain('animation: 5s host-a_foo ease;');
+    expect(result).toContain('animation: 3s .5s infinite steps(3,end) host-a_foo;');
+    expect(result).toContain('animation: 5s steps(9, jump-start) host-a_jump .5s;');
+    expect(result).toContain('animation: 1s step-end host-a_steps;');
+  });
+
+  it('should handle the usage of quotes', () => {
+    const css = `
+        div {
+            animation: 1.5s foo;
+        }
+
+        p {
+            animation: 1s 'foz bar';
+        }
+
+        @keyframes 'foo' {}
+        @keyframes "foz bar" {}
+        @keyframes bar {}
+        @keyframes baz {}
+        @keyframes qux {}
+        @keyframes quux {}
+        `;
+    const result = s(css, 'host-a');
+    expect(result).toContain('@keyframes \'host-a_foo\' {}');
+    expect(result).toContain('@keyframes "host-a_foz bar" {}');
+    expect(result).toContain('animation: 1.5s host-a_foo;');
+    expect(result).toContain('animation: 1s \'host-a_foz bar\';');
+  });
+
+  it('should handle the usage of quotes containing escaped quotes', () => {
+    const css = `
+        div {
+            animation: 1.5s "foo\\"bar";
+        }
+
+        p {
+            animation: 1s 'bar\\' \\'baz';
+        }
+
+        button {
+            animation-name: 'foz " baz';
+        }
+
+        @keyframes "foo\\"bar" {}
+        @keyframes "bar' 'baz" {}
+        @keyframes "foz \\" baz" {}
+        `;
+    const result = s(css, 'host-a');
+    expect(result).toContain('@keyframes "host-a_foo\\"bar" {}');
+    expect(result).toContain(`@keyframes "host-a_bar' 'baz" {}`);
+    expect(result).toContain(`@keyframes "host-a_foz \\" baz" {}`);
+    expect(result).toContain('animation: 1.5s "host-a_foo\\"bar";');
+    expect(result).toContain(`animation: 1s 'host-a_bar\\' \\'baz';`);
+    expect(result).toContain(`animation-name: 'host-a_foz " baz';`);
+  });
+
+  it('should handle the usage of commas in multiple animation definitions in a single declaration',
+     () => {
+       const css = `
+        button {
+        animation: 1s "foo bar, baz", 2s 'qux quux';
+        }
+
+        div {
+        animation: 500ms foo, 1s 'bar, baz', 1500ms bar;
+        }
+
+        p {
+        animation: 3s "bar, baz", 3s 'foo, bar' 1s, 3s "qux quux";
+        }
+
+        @keyframes "qux quux" {}
+        @keyframes "bar, baz" {}
+    `;
```

Comment:
Indentation is a little funky in this block (and the following tests) :grin:
Could we change it to something like:

```suggestion
       const css = `
         button {
           animation: 1s "foo bar, baz", 2s 'qux quux';
         }

         div {
           animation: 500ms foo, 1s 'bar, baz', 1500ms bar;
         }

         p {
           animation: 3s "bar, baz", 3s 'foo, bar' 1s, 3s "qux quux";
         }

         @keyframes "qux quux" {}
         @keyframes "bar, baz" {}
       `;
```

Reply:
sorry I keep messing up with the formatting.... :sweat: